### PR TITLE
Update to api 3.1

### PR DESCRIPTION
**Pull request by aneojgurhem**
Update gRPC API package to its latest version and implement missing calls in services.
Implements :
- Applications Service ListApplications
- Results Service ListResults
- Tasks Service CancelTaskAsync

Adds support for grpc web

fix https://github.com/aneoconsulting/ArmoniK/issues/514
fix https://github.com/aneoconsulting/ArmoniK/issues/366
fix https://github.com/aneoconsulting/ArmoniK/issues/525
fix https://github.com/aneoconsulting/ArmoniK/issues/526

### CODE REVIEW

File: Adaptors/Memory/src/TaskTable.cs [hunk 1]
```diff
@@ -180,11 +178,43 @@ public class TaskTable : ITaskTable
...
       .ConfigureAwait(false);
   }
 
+  /// <inheritdoc />
+  public Task<IList<TaskData>> CancelTaskAsync(ICollection<string> taskIds,
+                                               CancellationToken   cancellationToken = default)
+  {
+    foreach (var taskId in taskIds)
+    {
+      if (!taskId2TaskData_.TryGetValue(taskId,
```
[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Adaptors/Memory/src/TaskTable.cs:192]: Il ignore le task id en entrée s'il ne le trouve pas

File: Adaptors/Memory/src/TaskTable.cs [hunk 2]
```diff
@@ -180,11 +178,43 @@ public class TaskTable : ITaskTable
...
 
     await UpdateAllTaskStatusAsync(sessionFilter,
-                                   TaskStatus.Canceling,
+                                   TaskStatus.Cancelling,
                                    cancellationToken)
       .ConfigureAwait(false);
   }
 
+  /// <inheritdoc />
+  public Task<IList<TaskData>> CancelTaskAsync(ICollection<string> taskIds,
```
[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Adaptors/Memory/src/TaskTable.cs:187]: Il ne cancel pas la tâche directement, il met la tâche à cancelling.
Le polling agent acquiert la tache depuis la file d'attente, vois qu'elle est à cancelling puis la met à cancelled et enlève le message de la file d'attente.

File: Adaptors/MongoDB/src/SessionTable.cs [hunk 3]
```diff
@@ -238,25 +237,30 @@ await foreach (var sessionId in sessionCollection.AsQueryable(sessionHandle)
     }
   }
 
-  public async Task<IEnumerable<SessionData>> ListSessionsAsync(ListSessionsRequest request,
-                                                                CancellationToken   cancellationToken = default)
+  public async Task<(IEnumerable<SessionData> sessions, int totalCount)> ListSessionsAsync(Expression<Func<SessionData, bool>>    filter,
```
[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/SessionTable.cs:240]: oh si, il en manque beaucoup

**Review by jfonseca-aneo — APPROVED** [review]

File: Common/tests/ListSessionsRequestValidatorTest.cs [hunk 4]
```diff
@@ -134,4 +134,12 @@ public void ListSessionsRequestNegativePageSizeShouldFail()
     Assert.IsFalse(validator_.Validate(validListSessionsRequest_)
                              .IsValid);
   }
+
+  [Test]
+  public void ListSessionsZeroNegativePageSizeShouldFail()
+  {
+    validListSessionsRequest_!.PageSize = 0;
```
Comment (jfonseca-aneo) [inline Common/tests/ListSessionsRequestValidatorTest.cs:141]: I think you want a negative number here ^^

File: Adaptors/MongoDB/tests/MongoDatabaseProvider.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+  public void Dispose()
+  {
+    provider_.Dispose();
+    runner_?.Dispose();
+  }
+
+  public IServiceProvider GetServiceProvider()
+    => provider_;
+}
```
Comment (jfonseca-aneo) [inline Adaptors/MongoDB/tests/MongoDatabaseProvider.cs:125]: I think we could move this to the Test Helpers dir, the same code is used a least in the submitter tests

File: Common/tests/ListSessionsRequestValidatorTest.cs (hunk 4 above)
Comment (aneojgurhem) [reply Common/tests/ListSessionsRequestValidatorTest.cs:141]: The test has the wrong name^^ I have 2 tests a negative and a 0

File: Adaptors/MongoDB/tests/MongoDatabaseProvider.cs (hunk 5 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/tests/MongoDatabaseProvider.cs:125]: I've seen it in 2 other classes but the integration in those tests is more complicated than for those.

File: Common/tests/ListApplicationsRequestValidatorTest.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  [Test]
+  public void ListApplicationsRequestNegativePageSizeShouldFail()
+  {
+    validListApplicationsRequest_!.PageSize = -1;
+    Assert.IsFalse(validator_.Validate(validListApplicationsRequest_)
+                             .IsValid);
+  }
+
+  [Test]
+  public void ListApplicationsZeroNegativePageSizeShouldFail()
```
Comment (jfonseca-aneo) [inline Common/tests/ListApplicationsRequestValidatorTest.cs:139]: ZeroPageSizeShouldFail?

**Review by jfonseca-aneo — APPROVED** [review]

**Pull request merged**